tasks/ts.ts

```diff
@@ -664,22 +339,24 @@ function pluginFn(grunt: IGrunt) {
                     if (!!amdloaderPath) {
                         var referenceOrder: amdLoaderModule.IReferences
                             = amdLoaderModule.getReferencesInOrder(referenceFile, referencePath, generatedFiles);
-                        amdLoaderModule.updateAmdLoader(referenceFile, referenceOrder, amdloaderFile, amdloaderPath, rawTargetConfig.outDir);
+                        amdLoaderModule.updateAmdLoader(referenceFile, referenceOrder, amdloaderFile, amdloaderPath, currentFiles.outDir);
                     }
 
                     // Transform files as needed. Currently all of this logic in is one module
                     transformers.transformFiles(filesToCompile /*TODO: only unchanged files*/,
-                      filesToCompile, rawTargetConfig, options, lineEndingToUse);
+                      filesToCompile, options);
+
+                    currentFiles.src = filesToCompile;
 
                     // Return promise to compliation
                     if (options.compile) {
-                        // Compile, if there are any files to compile!
-                        if (filesToCompile.length > 0) {
-                            return runCompilation(filesToCompile, rawTargetConfig, options).then((success: boolean) => {
+                        if (filesToCompile.length > 0 || options.testExecute) {
+                            return runCompilation(options, currentFiles).then((success: boolean) => {
                                 return success;
                             });
                         }
-                        else { // Nothing to do
+                        else {
+                            // Nothing to do
                             grunt.log.writeln('No files to compile'.red);
```

Comment:
Same: this error will trigger in the tsconfig passThrough case. Perhaps another `||` clause for that case?